lib/uid_record.js

```diff
@@ -0,0 +1,95 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * This module keeps track of events related to a specific uid
+ */
+module.exports = function (limits, now) {
+
+  now = now || Date.now
+  function UidRecord() {
+    this.timestampsByAction = {}
+    this.rateLimitedTimestamp = undefined
```

Comment:
Even though it's going to make the stored JSON bigger, this is so much better with readable field names rather than abbreviations like `this.bk` and `this.rl`; very nice :-)